### PR TITLE
[v14] Create Umbraco/Bellissima Package RCL Template

### PR DESCRIPTION
**Pull request by leekelleher**
### Prerequisites

- [x] FYI @iOvergaard 👋 

### Description

During the Umbraco Community Teams days (2023), the Backoffice Team undertook the task to update the Umbraco Package template to work with the latest v14 preview release.

The updated template would be opinionated and install the following npm packages:

- `@umbraco-cms/backoffice`
- TypeScript
- Vite

The template would have an example **Dashboard** and a **Property Editor**.

Currently, this pull request is only for the RCL version of the template.  Going forwards, it is preferable to share the source files between both the RCL and non-RCL versions of the template, _(although that is currently beyond my knowledge of how do to that with `dotnet new` templates - any help appreciated)._ 🙏

### CODE REVIEW

File: templates/UmbracoPackageRcl/UmbracoPackage.csproj [hunk 1]
```diff
@@ -24,4 +24,21 @@
...
+
+  <ItemGroup>
+    <Content Update="package.json" Pack="False" />
+    <Content Update="package-lock.json" Pack="False" />
+    <Content Update="tsconfig.json" Pack="False" />
+  </ItemGroup>
+
+  <Target Name="NpmInstall" AfterTargets="Restore;Build" Inputs="package-lock.json" Outputs="node_modules\.package-lock.json">
+    <Message Text="Installing NPM packages" Importance="High" />
+    <Exec Command="npm ci --no-fund --no-audit --prefer-offline" />
```
Comment (JasonElkin) [inline templates/UmbracoPackageRcl/UmbracoPackage.csproj:36]: `npm ci` performs a clean install of all the packages, i.e. it wipes out and recreates `./node_modules/` - that shouldn't be necessary to run for every build. The ASP.NET example templates  add some conditional logic to only run when `/node_modules/` doesn't exist.

https://docs.npmjs.com/cli/v10/commands/npm-ci#description

**Comment by JasonElkin** [comment]
Excited to see this (but mainly leaving a comment so that the Collaborators dashboard doesn't shout at us 😁)

File: templates/UmbracoPackageRcl/UmbracoPackage.csproj (hunk 1 above)
Comment (iOvergaard) [reply templates/UmbracoPackageRcl/UmbracoPackage.csproj:36]: I'm wondering if it shouldn't be enough to run `npm install` which would make the incremental builds even faster. With modern npm, the discrepancies between the lock file and package file are less critical and you won't end up with a completely different version of a sub-sub-dependency anymore, as was the case of older npm, which spawned the need for the `ci` command in the first place. Also, if someone for some reason forgot to commit the changed lockfile, the `ci` command actually fails resulting in the entire build failing. This is not the case with `npm install`. What do you think?

**Comment by leekelleher** [comment]
Closing this PR, as it has been superseded by PR #17044.

**Pull request closed without merging**